### PR TITLE
refactor: Handle multiple equations

### PR DESCRIPTION
**Pull request by PapyChacal**
Basic refactoring to allow multiple equations in a timeloop.
Next step should be to properly do this as part of a DeVito Cluster, not just assuming a bunch of equations are supposed to be lowered this way always!

Also:
- Use the operator name in the generated IR.

### CODE REVIEW

File: devito/ir/ietxdsl/cluster_to_ssa.py [hunk 1]
```diff
@@ -120,41 +118,45 @@ def _convert_eq(self, eq: LoweredEq):
...
         # Also, store the output function's time buffer of this equation
         output_time_offset = (eq.lhs.indices[step_dim] - step_dim) % eq.lhs.function.time_size
         self.out_time_buffer = (output_function, output_time_offset)
 
-        # For each used time_buffer, define a stencil.field type for the function .
-        fields_types = [field_from_function(f) for (f, _) in self.time_buffers]
+        # Bounds and step boilerpalte
+        lb = iet_ssa.LoadSymbolic.get(step_dim.symbolic_min._C_name, builtin.IndexType())
+        ub = iet_ssa.LoadSymbolic.get(step_dim.symbolic_max._C_name, builtin.IndexType())
+        one = arith.Constant.from_int_and_width(1, builtin.IndexType())
```
Comment (georgebisbas) [inline devito/ir/ietxdsl/cluster_to_ssa.py:135]: TODO: we will parse this

File: devito/ir/ietxdsl/cluster_to_ssa.py [hunk 2]
```diff
@@ -120,41 +118,45 @@ def _convert_eq(self, eq: LoweredEq):
...
+        try:
+            step = arith.Constant.from_int_and_width(
```
Comment (georgebisbas) [inline devito/ir/ietxdsl/cluster_to_ssa.py:137]: Actually l.135 is uselss

File: devito/ir/ietxdsl/cluster_to_ssa.py [hunk 3]
```diff
@@ -120,41 +118,45 @@ def _convert_eq(self, eq: LoweredEq):
...
+            )
+            step.result.name_hint = "step"
+        except:
+            raise ValueError("step must be int!")
 
-        # Create a function with the fields as arguments
-        xdsl_func = func.FuncOp("apply_kernel", (fields_types, []))
+        iter_args = list(self.function_args.values())
+        # Create the for loop
+        loop = scf.For(lb, arith.Addi(ub, one), step, iter_args, Block(arg_types=[builtin.IndexType(), *(a.type for a in iter_args)]))
```
[the remark this answers is not in the document]
Comment (PapyChacal) [reply devito/ir/ietxdsl/cluster_to_ssa.py]: Step is step and is not hardcoded
The hardcoded one is only to iterate from time_m to time_M+1

File: devito/ir/ietxdsl/cluster_to_ssa.py (hunk 2 above)
Comment (PapyChacal) [reply devito/ir/ietxdsl/cluster_to_ssa.py]: Idem, not useless, just used to defined time_M+1
There are probably elegant alternatives, but I'd say it's no priority at this point?

**Pull request merged**